workspaces/scorecard/plugins/scorecard-backend-module-jira/src/clients/base.ts

```diff
@@ -0,0 +1,184 @@
+/*
+ * Copyright Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import type { Config } from '@backstage/config';
+import type { Entity } from '@backstage/catalog-model';
+import {
+  JiraConfig,
+  JiraEntityFilters,
+  JiraOptions,
+  Product,
+  RequestOptions,
+} from '../types';
+import {
+  ANNOTATION_JIRA_PROJECT_KEY,
+  ANNOTATION_JIRA_COMPONENT,
+  ANNOTATION_JIRA_LABEL,
+  ANNOTATION_JIRA_TEAM,
+  ANNOTATION_JIRA_CUSTOM_FILTER,
+  API_VERSION_DEFAULT,
+  JIRA_CONFIG_PATH,
+  JIRA_OPTIONS_PATH,
+  JIRA_MANDATORY_FILTER,
+  CONFIG_TOKEN,
+  CONFIG_PRODUCT,
+  CONFIG_API_VERSION,
+  CONFIG_BASE_URL,
+  CONFIG_MANDATORY_FILTER,
+  CONFIG_CUSTOM_FILTER,
+} from '../constants';
+
+export abstract class JiraClient {
+  protected readonly config: JiraConfig;
+  protected readonly options?: JiraOptions;
+
+  constructor(config: Config) {
+    const jiraConfig = config.getConfig(JIRA_CONFIG_PATH);
+    this.config = {
+      baseUrl: jiraConfig.getString(CONFIG_BASE_URL),
+      token: jiraConfig.getString(CONFIG_TOKEN),
+      product: jiraConfig.getString(CONFIG_PRODUCT) as Product,
+      apiVersion: jiraConfig.getOptionalString(CONFIG_API_VERSION),
+    };
+
+    const jiraOptions = config.getOptionalConfig(JIRA_OPTIONS_PATH);
+    if (jiraOptions) {
+      this.options = {
+        mandatoryFilter: jiraOptions.getOptionalString(CONFIG_MANDATORY_FILTER),
+        customFilter: jiraOptions.getOptionalString(CONFIG_CUSTOM_FILTER),
+      };
+    }
```

Comment:
As discussed today in our short call. Maybe you can reuse the Config (from config.d.ts) as a type instead of reading all individual config values.

Small example:

```tsx
type JiraConfig = NonNullable<NonNullable<NonNullable<Config['scorecard']>['plugins']>['jira']>['options'];

// you should get it then via:
rootConfig.get("jira").get<JiraConfig>();
```

Or?